compute_tools/src/bin/compute_ctl.rs

```diff
@@ -202,6 +222,47 @@ fn main() -> Result<()> {
         }
     };
 
+    let result = ProcessCliResult {
+        // directly from CLI:
+        connstr,
+        pgdata,
+        pgbin,
+        ext_remote_storage,
+        http_port,
+        // others:
+        spec,
+        live_config_allowed,
+    };
+
+    // TODO: Move startup_context_guard out of this function. It's here right now only because
+    // that's where it was before, and moving it would've made the diff big.
+    Ok((startup_context_guard, result))
+}
+
+struct ProcessCliResult<'clap> {
```

Comment:
Doesn't clap have a way to to parse a command line into a struct? Seems like that would be the best solution to cleaning up the command line code, imo.

Reply:
Yup- I quite like its derive API. I figured that's a non-trivial change, only want to tackle that in this PR if there's +1s from compute team.

Reply:
I am definitely a +1 on it :), but you also don't need to tackle it as part of this PR. I could get to it.